experiments/scripts/env_creation_hf.sh

```diff
@@ -8,21 +8,29 @@ export CONDA_ENV_PATH=/fsx/proj-chemnlp/$1/conda/env/chemnlp-hf
 export PYTHON_VER=3.8
 CUDA_VERSION=11.7
 CONDA_BASE=$(conda info --base)
+CHEMNLP_PATH=/fsx/proj-chemnlp/$2/chemnlp
 
 ## ensure we can use activate syntax in slurm scripts
 source $CONDA_BASE/etc/profile.d/conda.sh
 
 # Create Python environment through conda
-if [ -d "${CONDA_ENV_PATH}" ]; then rm -Rf ${CONDA_ENV_PATH}; fi
-conda create --force --prefix ${CONDA_ENV_PATH} python=${PYTHON_VER} -y
-conda activate ${CONDA_ENV_PATH}
+if [ -d "${CONDA_ENV_PATH}" ]
+then
+    # if already exists activate
+    echo "Found ${CONDA_ENV_PATH} in the directory, activating it!"
+    conda activate ${CONDA_ENV_PATH}
+else
+    # otherwise create new env (race conditions exist)
+    echo "Creating ${CONDA_ENV_PATH} environment!"
+    conda create --force --prefix ${CONDA_ENV_PATH} python=${PYTHON_VER} -y
+    conda activate ${CONDA_ENV_PATH}
 
-# Python requirements
-## cd into your directory inside of proj-chemnlp
-cd /fsx/proj-chemnlp/$2
+    ## clone + submodules (ok if exists)
+    cd /fsx/proj-chemnlp/$2
+    [ ! -d 'chemnlp' ] && git clone --recurse-submodules git@github.com:OpenBioML/chemnlp.git
 
-## clone + submodules (ok if exists)
-[ ! -d 'chemnlp' ] && git clone --recurse-submodules git@github.com:OpenBioML/chemnlp.git
-
-## install core requirements
-conda install -y pytorch torchvision torchaudio pytorch-cuda=${CUDA_VERSION} -c pytorch -c nvidia --verbose
+    ## install core requirements
+    conda install -y pytorch torchvision torchaudio pytorch-cuda=${CUDA_VERSION} -c pytorch -c nvidia --verbose
+    cd $CHEMNLP_PATH
+    pip install ".[training]"
+fi
```

Comment:
what is fi?

Reply:
it closes the `if` statement so that any lines below that aren't considered part of the `else` clause and are just new statements.